tsconfig.json

```diff
@@ -0,0 +1,20 @@
+{
```

Comment:
Overall, I like the idea of introducing TypeScript to this project, but if we want to do that, I think we need to migrate all `.js` files into TypeScript at some point, otherwise, we will end up with an ugly mix of JavaScript, and Typescript :D

Reply:
Sure, I don't think we would need to have a bi-lingual codebase for too long. With the current `tsconfig` there are 70 type errors over 31 files, it's not such a big lift to fix those in 1-2 PRs. Then we can switch everything to TS and gradually enable stricter type-checking rule by rule.

Reply:
One of the points to have js code was to have js codebase as an example for customers who don't know TypeScript. But with TS codebase we can compile it to js and get code in 2 languages.

I'm not sure that after fixing 70 errors we wouldn't get 300 new ones, but I like this initiative!

Reply:
I think if you go easy on the types (avoid needless generics, type props and function params, infer everything you can) you get TS files that are 90-95% identical to JS files and reasonably easy to navigate and read even for people who only know JavaScript. 

You are not expecting JS devs to *write* TS, only to read it and types are easier to delete/ignore than to add :) 

And yes, you can always compile it to ES6+ JS and provide that to people who prefer JS. The effort to do that is easily outweighed by the TypeScript productivity boost.